src/main/java/io/zipcoder/Classroom.java

```diff
@@ -73,15 +151,69 @@ public double getClassAverage(){
      * @return
      */
     public String getClassScores(){
-        return null;
+        if (this.studentsEnrolled == 0) return "No Students";
+        StringBuilder finalOutput = new StringBuilder("Students:\n");
+        finalOutputAssembler(finalOutput);
+        return finalOutput.toString();
+    }
+
+    private void finalOutputAssembler(StringBuilder finalOutput) {
+        for (Student validStudent : this.students){
+            if (validStudent != null){
+                finalOutput.append(validStudent.getFirstName() + " " + validStudent.getLastName() + " -> "
+                        + validStudent.getAverage() + "\n");
+            }
+        }
```

Comment:
this is another example of two methods, when one would do after you've figured out how to structure it. if you have Qs, come ask.